indexer/contacts_processor.cpp

```diff
@@ -0,0 +1,266 @@
+#include "indexer/contacts_processor.hpp"
+
+#include "base/logging.hpp"
+#include "base/string_utils.hpp"
+
+#include "coding/url.hpp"
+
+#include "indexer/editable_map_object.hpp"
+
+using namespace std;
+
+namespace osm {
+
+string ValidateAndFormat_facebook(string const & facebookPage)
+{
+  if (facebookPage.empty())
+    return {};
+  // Check that facebookPage contains valid username. See rules: https://www.facebook.com/help/105399436216001
+  if (strings::EndsWith(facebookPage, ".com") || strings::EndsWith(facebookPage, ".net"))
+    return {};
+  if (regex_match(facebookPage, regex(R"(^@?[a-zA-Z\d.\-]{5,}$)")))
+  {
+    if (facebookPage.front() == '@')
+      return facebookPage.substr(1);
+    else
+      return facebookPage;
+  }
+  if (EditableMapObject::ValidateWebsite(facebookPage))
+  {
+    string facebookPageUrl = facebookPage;
+    // Check if HTTP protocol is present
+    if (!strings::StartsWith(facebookPage, "http://") && !strings::StartsWith(facebookPage, "https://"))
+      facebookPageUrl = "https://" + facebookPage;
+
+    const url::Url url = url::Url(facebookPageUrl);
+    const string & domain = strings::MakeLowerCase(url.GetWebDomain());
```

Comment:
1. If a function returns an object, there is no benefit of using const & reference to it (compiler creates and keeps a full string anyway). Just use `string const`.
2. nit: use right-to-left types everywhere, according to our style guide: "a reference to a constant string" is `string const &`

Reply:
Fixed